django/db/models/expressions.py

```diff
@@ -319,6 +323,59 @@ def get_source_fields(self):
         """
         return [e._output_field_or_none for e in self.get_source_expressions()]
 
+    def asc(self):
+        return OrderBy(self)
+
+    def desc(self):
+        return OrderBy(self, descending=True)
+
+    def reverse_ordering(self):
+        return self
+
+
+class ExpressionNode(BaseExpression, CombinableMixin):
+    """
+    An expression that can be combined with other expressions.
+    """
+    pass
+
+
+class OrderBy(BaseExpression):
+    template = '%(expression)s %(ordering)s'
+    descending_template = 'DESC'
+
+    def __init__(self, expression, descending=False, nulls_first=False, nulls_last=False):
+        self.descending = descending
+        self.nulls_first = nulls_first
+        self.nulls_last = nulls_last
+        if not hasattr(expression, 'resolve_expression'):
+            raise ValueError('expression must be an expression type')
+        self.expression = expression
+
+    def set_source_expressions(self, exprs):
+        self.expression = exprs[0]
+
+    def get_source_expressions(self):
+        return [self.expression]
+
+    def as_sql(self, compiler, connection):
+        expression_sql, params = compiler.compile(self.expression)
+        placeholders = {'expression': expression_sql}
+        placeholders['ordering'] = self.descending_template if self.descending else ''
+        return (self.template % placeholders).rstrip(), params
+
+    def reverse_ordering(self):
+        self.descending = not self.descending
+        self.nulls_first = not self.nulls_first
+        self.nulls_last = not self.nulls_last
+        return self
+
+    def asc(self):
```

Comment:
Is there any reason why these methods can't just invert `self.descending`? I agree `Expression().asc().desc()` looks weird, but it is deterministic (should be descending).